### ISSUE TITLE
[v2.9+quilt-1.19.1] Statistics dont update after rejoin

### ISSUE THREAD

**Issue by Rozbiynk** [TheCSMods/mc-better-stats#60]
### What happened?

![изображение](https://github.com/TheCSMods/mc-better-stats/assets/95433246/dd89512c-b640-4292-ada3-527c5f385ddc)
sp2
![изображение](https://github.com/TheCSMods/mc-better-stats/assets/95433246/ac2d27a9-37ba-4071-9099-d64132170b62)
sp1
![изображение](https://github.com/TheCSMods/mc-better-stats/assets/95433246/c230bf83-ea9e-426f-bd0c-8e790cc3aacd)
mp lobby (had its own statistics)
![изображение](https://github.com/TheCSMods/mc-better-stats/assets/95433246/b9993947-f1f0-4f59-9af8-bfaf329ef6f6)
mp vanilla 

### Steps to reproduce

1. Join to mp
2. Join sp1
3. Join sp2 and see statistic from sp1
TESTED WITHOUT MOD, without it everything is okay

### Relevant log output or crash report

```shell
https://mclo.gs/4uccbTJ
```

### Other installed mods

https://www.mediafire.com/file/4h06hqe21q3kcfl/Lantern.zip/file (same link)

### Code of Conduct

- [X] I agree that if the issue is regarding a crash, I gave the logs and the crash report.
- [X] I agree that I will be available later for any follow-up questions to help diagnose and resolve the issue.

**Comment by TheCSDev** [comment]
Ah, a non-rendering-related bug this time. I thought I implemented an event handler to clear the stats when leaving a world, so they can update on another world. I guess I completely forgot to implement that in the end.  
  
Thanks for spotting that!  
Come to think of it, you hunted down quite some bugs today, so I have a question.  
  
Would you be interested in being granted the [bug hunter badge](https://github.com/TheCSMods/mc-better-stats/wiki/Player-Badge-system#-better-stats-bug-hunter) ([please read the 'Important' paragraph here](https://github.com/TheCSMods/mc-better-stats/wiki/Player-Badge-system#this-repositorys-badges))?  
(keep in mind those are permanently public btw)

**Comment by TheCSDev** [comment]
Reopening until it's published.

**Comment by Rozbiynk** [comment]
yes, i interested in badge, and it that time i working on translate 2.9.1)

**Comment by TheCSDev** [comment]
Release published. Will take some time for CurseForge approves it on their end. In the meanwhile, I hope there are no more bugs as bad as that one remaining, as I'm hoping to get to 1.20 as soon as possible.
